README.md

```diff
@@ -1,2 +1,485 @@
+
+
+- Output
+
+  This run profile executes a single test on an internally hosted web application on a single browser on BrowserStack. Please refer to your BrowserStack dashboard(https://automate.browserstack.com/) for test results.
+
+
+### [Web application hosted on internal environment] Run the entire test suite in parallel on a single BrowserStack browser using BrowserStackLocal
+
+In this section, we will run the test cases to test the internally hosted website in parallel on a single browser on Browserstack. Refer to the `single_local` object in `caps.json` file to change test capabilities for this configuration.
+
+- How to run the test?
+
+  To run the entire test suite in parallel on a single BrowserStack browser using BrowserStackLocal, use the following command:
+  Maven:
+  ```sh
+  mvn test -P bstack-local-parallel
+  ```
+
+  Gradle:
+  ```sh
+  gradle bstack-local-parallel
+  ```
+
+- Output
+
+  This run profile executes the entire test suite on an internally hosted web application on a single browser on BrowserStack. Please refer to your [BrowserStack dashboard](https://automate.browserstack.com/) for test results.
+
+- Note: By default, this execution would run maximum 5 test threads in parallel on BrowserStack. Refer to the section ["Configuring the maximum parallel test threads for this repository"](#Configuring-the-maximum-parallel-test-threads-for-this-repository) for updating the parallel thread count based on your requirements.
+
+### [Web application hosted on internal environment] Run the entire test suite in parallel on multiple BrowserStack browser using BrowserStackLocal
+
+In this section, we will run the test cases to test the internally hosted website in parallel on multiple browsers on Browserstack. Refer to the `parallel_local` object in `caps.json` file to change test capabilities for this configuration.
+
+- How to run the test?
+
+  To run the entire test suite in parallel on a single BrowserStack browser using BrowserStackLocal, use the following command:
+
+  Maven:
+  ```sh
+  mvn test -P bstack-local-parallel-browsers
+  ```
+
+  Gradle:
+    ```sh
+  gradle bstack-local-parallel-browsers
+  ```
+
+- Output
+
+  This run profile executes the entire test suite on an internally hosted web application on multiple browsers on BrowserStack. Please refer to your [BrowserStack dashboard](https://automate.browserstack.com/) for test results.
+
+- Note: By default, this execution would run maximum 5 test threads in parallel on BrowserStack. Refer to the section ["Configuring the maximum parallel test threads for this repository"](#Configuring-the-maximum-parallel-test-threads-for-this-repository) for updating the parallel thread count based on your requirements.
+
+## Generating Allure Reports
+
+- Generate Report using the following command: `mvn allure:generate`
+- Serve the Allure report on a server: `mvn allure:serve`
```

Comment:
There is a "known issues" section in the template readme. Mention "Allure with Gradle has not yet been configured and we are working on it." if Allure is not working with Gradle. We will pick it up in next iteration / release. If it is working, add the gradle equivalent commands